x-pack/plugins/inference/server/inference_client/inference_client.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger } from '@kbn/logging';
+import type { KibanaRequest } from '@kbn/core-http-server';
+import type { PluginStartContract as ActionsPluginStart } from '@kbn/actions-plugin/server';
+import type { InferenceClient } from './types';
+import { createChatCompleteApi } from '../chat_complete';
+import { createOutputApi } from '../../common/output/create_output_api';
+import { getConnectorById } from '../util/get_connector_by_id';
+
+export function createInferenceClient({
```

Comment:
question/optional nit: If I follow the code paths correctly, this only appears to be used within `./create_client` (and its test). Is this extracted to facilitate mocking/testing, or is there another benefit to having this in its own file?